src/components/BaseTable.vue

```diff
@@ -0,0 +1,121 @@
+<template>
+	<BaseCard margin="sm">
+		<table class="base-table">
+			<BaseTableHeader
+				:headers="headers"
+				:sort-by="sortBy"
+				@sort-by-changed="onSortByChanged"
+			/>
+
+			<slot :items="pagedData" />
+		</table>
+	</BaseCard>
+
+	<div class="base-table__footer">
+		<BaseTablePagination
+			:page-size="pageSize"
+			:current-page="currentPage"
+			:items-count="itemsCount"
+			@page-size-changed="onPageSizeChanged"
+			@page-changed="onPageChanged"
+		/>
+	</div>
+</template>
+
+<script setup>
+import BaseCard from './BaseCard'
+import BaseTableHeader from './BaseTableHeader'
+import BaseTablePagination from './BaseTablePagination'
+import { computed, ref, watch } from 'vue'
+
+const props = defineProps({
+	headers: {
+		type: Object,
+		required: true
+	},
+
+	initialSortBy: {
+		type: String,
+		default: ''
+	},
+
+	items: {
+		type: Array,
+		required: true
+	}
+})
+
+const pageSize = ref(10)
+const currentPage = ref(0)
+
+const sortBy = ref(props.initialSortBy)
+const sortDesc = ref(true)
+
+const pagedData = computed(() => {
+	const startIndex = currentPage.value * pageSize.value
+	const endIndex = (currentPage.value + 1) * pageSize.value
+	return sortedItems.value.slice(startIndex, endIndex)
+})
+
+const itemsCount = computed(() => sortedItems.value.length)
+
+watch(pageSize, () => (currentPage.value = 0))
+
+const sortedItems = computed(() => {
+	const sortKey = sortBy.value
+	const sortDescValue = sortDesc.value
+
+	const itemsCopy = [...props.items]
+
+	return itemsCopy.sort((a, b) => {
+		if (a[sortKey] === b[sortKey]) {
+			return 0
+		}
+
+		if (sortDescValue) {
+			if (typeof a[sortKey] === 'number') {
+				return b[sortKey] - a[sortKey]
+			} else {
+				return a[sortKey] > b[sortKey] ? -1 : 1
+			}
+		} else {
+			if (typeof a[sortKey] === 'number') {
+				return a[sortKey] - b[sortKey]
+			} else {
+				return a[sortKey] < b[sortKey] ? -1 : 1
+			}
+		}
+	})
```

Comment:
we have lodash  in this repo, so you can
`import { orderBy } from 'lodash'` 
and then just do
```suggestion
        const sortDir = sortDesc.value ? 'desc' : 'asc'
        return orderBy(props.items, sortBy.value, sortDir)
```

Reply:
I disagree here, I was looking at where it is used in `ui-common-library` and I can't find a single usage.

I see it is used in:
`holofuel-rsm-ui` - 2x `get` / 1x `pick` / 1x `escapeRegExp` occurrences
`publisher-portal` - 2x `pick` occurrences
`host-console-ui` - 2x `isFunction` / 1x `isEmpty` (I just replaced them with native JS code and ditched `lodash` from the project)

I think `lodash` is also slowly becoming obsolete, unless we are really doing some crazy data manipulations, but I don't think we are. Most of the things `lodash` gives us can be achieved by using modern JS which is x times faster.

I am not very familiar with `lodash` as I am too young as a developer I guess. I was looking at what lodash is actually giving us and found this https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore, that opened my eyes a bit.

### USE NATIVE JS
I understand why it was used in the past, eg. `pick`, I see that doing that pre ES6 destructuring was madness, but from what I understand all it does is actually destructuring, that can be achieved very easily now.

https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#_pick

```
var object = { 'a': 1, 'b': '2', 'c': 3 };

// Underscore/Lodash
var result = _.pick(object, ['a', 'c']);
console.log(result)
// output: {a: 1, c: 3}

// Native (destructuring)
const { a, c } = object;
const result = { a, c};
console.log(result);
// output: {a: 1, c: 3}
// for an array of this object --> array.map(({a, c}) => ({a, c}));

// Native
function pick(object, keys) {
  return keys.reduce((obj, key) => {
     if (object && object.hasOwnProperty(key)) {
        obj[key] = object[key];
     }
     return obj;
   }, {});
}
var result = pick(object, ['a', 'c']);
console.log(result)
// output: {a: 1, c: 3}
```

Back to my `.sort`, I just checked and I can simplify the solution by not checking the `typeof a[sortKey] === 'number'` I think something different was wrong with my data before.

So doing below works fine

```
return itemsCopy.sort((a, b) => {
  if (a[sortKey] === b[sortKey]) {
    return 0
  }

  if (sortDirection.value === ESortDirections.desc) {
    return a[sortKey] > b[sortKey] ? -1 : 1
  } else {
    return a[sortKey] < b[sortKey] ? -1 : 1
  }
})
```

### SUMMARY:
I don't think `lodash` is completely obsolete but I think we should use it only when we can't really find a native solution quickly or it becomes some kind of monster code with a lot of edge cases that need to be covered, then people from `lodash` probably solved it for us.

Also I think we should remove it from `ui-common-library` as it is not used and import it only in projects that really need that.